### PR TITLE
[FEATURE] Router

### PR DESCRIPTION
**Pull request by ThomasStinkensPXL**
Added a new router that redirect the users to the correct page (login, todos or 404error). Also made sure a user it authenticated when necessary (beforeEach in the router).

The way I check if a user is authenticated is by always getting the user info after a succesful login, this garuantees that currentUser is no longer null after a login.

### CODE REVIEW

File: src/modules/auth/views/AuthLoginView.vue [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<template>
+	<p>Login</p>
+</template>
+
+<script>
+export default {
```
Comment (maltsavkiryl) [inline src/modules/auth/views/AuthLoginView.vue]: deze export default is niet nodig :)

File: src/views/Error404View.vue [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<template>
+	<h1>404 Not Found</h1>
+</template>
+
+<script>
+export default {
+	name: 'Error404View',
```
Comment (maltsavkiryl) [inline src/views/Error404View.vue]: export default is niet nodig :)

File: src/views/TodoOverviewView.vue [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<template>
+	<p>Overview!</p>
+</template>
+
+<script>
+export default {
```
Comment (maltsavkiryl) [inline src/views/TodoOverviewView.vue]: hier mag ie ook weg :)

**Review by maltsavkiryl — CHANGES_REQUESTED** [review]

File: src/modules/auth/views/AuthLoginView.vue (hunk 1 above)
Comment (JeroenVanC) [reply src/modules/auth/views/AuthLoginView.vue]: <script setup> van maken

File: src/modules/auth/views/AuthLoginView.vue [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+	name: 'AuthLoginView',
+}
+</script>
+
+<style scoped>
```
Comment (JeroenVanC) [inline src/modules/auth/views/AuthLoginView.vue]: style mag ook weg !

Comment (JeroenVanC) [reply src/modules/auth/views/AuthLoginView.vue]: altijd tailwind gebruiken

**Review by JeroenVanC — CHANGES_REQUESTED** [review]

**Review by maltsavkiryl — APPROVED** [review]

**Pull request merged**